### PR TITLE
Add e2e tests for modular upgrades

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
E2E tests for modular upgrades. 

This verifies that upgrading the mgmt cluster EKS-A minor version does not affect workload custers

*Testing (if applicable):*
Ran docker E2E

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/side_effects.go [hunk 1]
```diff
@@ -203,3 +204,50 @@ func newEKSAPackagedBinaryForLocalBinary(tb testing.TB) eksaPackagedBinary {
 		version: version,
 	}
 }
+
+func runTestManagementClusterUpgradeSideEffects(t *testing.T, provider framework.Provider, mgmtFiller api.ClusterConfigFiller, workloadFillers ...api.ClusterConfigFiller) {
```
[the remark this answers is not in the document]
Comment (taneyland) [reply test/e2e/side_effects.go:208]: Yeah we would have to add some of the methods like `WithKubeVersion` and `WithWorkerNodeGroup` to the framework.Provider interface
https://github.com/aws/eks-anywhere/blob/main/test/framework/cluster.go#L311

I considered doing this while implementing but I was on the fence on whether or not to make this interface aware of these extra functions. But I don't have any strong objections against adding it. It'll definitely help clean up the method signature. What do you think? @g-gaston

[the remark this answers is not in the document]
Comment (g-gaston) [reply test/e2e/side_effects.go:208]: Yeah, I think that addition to the interface makes sense. And this puts in a path where more tests can be used cross provider.

just for my understanding, why is `WithWorkerNodeGroup` needed in the interface?

File: test/framework/vsphere.go [hunk 2]
```diff
@@ -503,6 +503,17 @@ func (v *VSphere) ClusterConfigUpdates() []api.ClusterConfigFiller {
 	return []api.ClusterConfigFiller{api.ClusterToConfigFiller(f...), api.VSphereToConfigFiller(v.fillers...)}
 }
 
+// WithKubeVersionAndOS returns a cluster config filler that sets the cluster kube version.
```
Comment (g-gaston) [inline test/framework/vsphere.go:506]: nit: update this comment :)

File: test/framework/tinkerbell.go [hunk 3]
```diff
@@ -153,6 +153,20 @@ func (t *Tinkerbell) CleanupVMs(_ string) error {
 	return nil
 }
 
+// WithKubeVersionAndOS returns a cluster config filler that sets the cluster kube version and the right template for all
+// tinkerbell machine configs.
+func (t *Tinkerbell) WithKubeVersionAndOS(osFamily anywherev1.OSFamily, kubeVersion anywherev1.KubernetesVersion) api.ClusterConfigFiller {
+	// TODO: Update tests to use this
```
Comment (g-gaston) [inline test/framework/tinkerbell.go:159]: we can maybe panic here?

**Review by g-gaston — APPROVED** [review]
amazing!

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Pull request merged**